app-text/aobook/aobook-1.0.3.recipe

```diff
@@ -4,54 +4,62 @@ of Japanese-language literature."
 HOMEPAGE="https://osdn.net/projects/aobook/"
 COPYRIGHT="2014-2017 Azel"
 LICENSE="BSD (3-clause)"
-REVISION="2"
+REVISION="3"
 localRevision="3"
 SOURCE_URI="https://github.com/hanya/aobook-haiku/archive/v$portVersion-$localRevision.tar.gz"
 CHECKSUM_SHA256="9b39981a204e6094563c2435979da4eea1aa2ef3d1a5e9580961a4e9812d34a7"
 SOURCE_DIR="aobook-haiku-$portVersion-$localRevision"
 
 ARCHITECTURES="!x86_gcc2 x86 x86_64"
+SECONDARY_ARCHITECTURES="x86"
 
 PROVIDES="
-	aobook = $portVersion
+	aobook$secondaryArchSuffix = $portVersion
 	app:Aobook = $portVersion
 	"
 REQUIRES="
-	haiku
-	lib:libfontconfig
-	lib:libfreetype
-	lib:libiconv
-	lib:libjpeg
-	lib:libpng
-	lib:libz
+	haiku$secondaryArchSuffix
+	lib:libfontconfig$secondaryArchSuffix
+	lib:libfreetype$secondaryArchSuffix
+	lib:libiconv$secondaryArchSuffix
+	lib:libjpeg$secondaryArchSuffix
+	lib:libpng16$secondaryArchSuffix
+	lib:libz$secondaryArchSuffix
 	"
 BUILD_REQUIRES="
-	haiku_devel
-	devel:libfontconfig
-	devel:libfreetype
-	devel:libiconv
-	devel:libjpeg
-	devel:libpng
-	devel:libz
+	haiku${secondaryArchSuffix}_devel
+	devel:libfontconfig$secondaryArchSuffix
+	devel:libfreetype$secondaryArchSuffix
+	devel:libiconv$secondaryArchSuffix
+	devel:libjpeg$secondaryArchSuffix
+	devel:libpng16$secondaryArchSuffix
+	devel:libz$secondaryArchSuffix
 	"
 BUILD_PREREQUIRES="
 	makefile_engine
-	cmd:g++
-	cmd:gcc
+	cmd:g++$secondaryArchSuffix
+	cmd:gcc$secondaryArchSuffix
 	cmd:make
+	cmd:sed
 	"
 
 USER_SETTINGS_FILES="
 	settings/aobook directory
 	"
 
+PATCH()
+{
+	sed -i -e "s,\(shell findpaths -e \)\(B_FIND_PATH_HEADERS_DIRECTORY freetype2)\),\1-a \$(ARCH) \2," \
+		mlib/Makefile.haiku src/Makefile.haiku
+}
+
 BUILD()
 {
 	cd mlib
-	make -f Makefile.haiku $jobArgs OBJ_DIR=objects
+	make -f Makefile.haiku $jobArgs OBJ_DIR=objects ARCH=$effectiveTargetArchitecture
 
 	cd ../src
```

Comment:
The recipe builds fine and everything seems to work fine (although I did not load any file with the app.)
So I we can merge your PR.

But before doing so, I'd like to mention something a noticed when I tried a variant for `BUILD()`:
```bash
BUILD()
{
	make -C mlib -f Makefile.haiku $jobArgs OBJ_DIR=objects \
		ARCH=$effectiveTargetArchitecture
	make -C src -f Makefile.haiku $jobArgs OBJ_DIR=objects \
		ARCH=$effectiveTargetArchitecture default catalogs bindcatalogs
}
```
What I noticed is that with this variant the app as bigger than when I build it with the recipe in this PR.
When I build aobook on x86 secondary arch with my variant, /system/apps/Aobook is around 526 KiB, instead of 490 KiB.
It's a bit puzzling, but I have to admit that I did not investigate a lot :)

Anyway, I can merge your PR as soon as you wish.

Reply:
I have to pass the architecture when catalogs is build to make freetype header is found. 
The difference of the binary file is the fail of the binding the catalog file generated by the catalogs.

Reply:
Cool, thanks for the investigation!
So, this means my last variant was a good idea :)

BTW, I've just tested one more variant (based on this one) which seems to be working fine.
You can drop the 2 ` ARCH=$effectiveTargetArchitecture` in `BUILD` and the whole `PATCH` function if you apply these small changes upstream, in `mlib/Makefile.haiku` and `src/Makefile.haiku`:
```diff
 LIBPATHS = 
 
+ARCH = $(shell getarch)
+
 #	Additional paths to look for system headers. These use the form
 #	"#include <header>". Directories that contain the files in SRCS are
 #	NOT auto-included here.
 SYSTEM_INCLUDE_PATHS = \
-	$(shell findpaths -e B_FIND_PATH_HEADERS_DIRECTORY freetype2) \
+	$(shell findpaths -e -a $(ARCH) B_FIND_PATH_HEADERS_DIRECTORY freetype2) \
```

Shall we merge the recipe as is, or would you prefer to implement this "last minute" variant before the merge?
(It's perfectly OK to merge this PR as is and postpone this last minute suggestion to a future release. Both scenari are perfectly OK.)